src/server/routes/calendar.js

```diff
@@ -59,20 +59,20 @@ router.post('/addevent', function(req,res){
   if(!google_calendar) {
     var google_calendar = new gcal.GoogleCalendar(req.user.calAccessToken);
   }
-
+  console.log(req.body);
   // Parse JSON
-  var eventJSON = JSON.parse(req.body.eventInfo);
+  
   // Insert Event into Google Database
   // Logic to Associate Calendar Summary with ID
   google_calendar.calendarList.list(function(err, calendarList) {
 
     for(var i = 0; i < calendarList.items.length; i++){
-      if(eventJSON.calendar === calendarList.items[i].summary){
+      if(req.body.calendar === calendarList.items[i].summary){
```

Comment:
There is no else. This is taking the calendarName that the user selected when adding the event and using the response from the API call to match with the calendar ID that is required to then send back the event to Google. I have optimized in case there is an error though but only from the error retrieve CalendaList API Call. That is the only reason the if that you highlighted shouldn't work.